gnosis/safe/safe.py

```diff
@@ -115,6 +108,26 @@ def __init__(self, address: ChecksumAddress, ethereum_client: EthereumClient):
     def __str__(self):
         return f"Safe={self.address}"
 
+    @cached_property
+    def contract(self) -> Contract:
+        v_1_3_0_contract = get_safe_V1_3_0_contract(self.w3, address=self.address)
+        version = v_1_3_0_contract.functions.VERSION().call()
+        if version == "1.3.0":
+            return v_1_3_0_contract
+        else:
+            return get_safe_V1_1_1_contract(self.w3, address=self.address)
```

Comment:
One question about this, why we return the contract 1.1.1 when for example the contract version is 1.2.0?

Reply:
Because they share "almost" the same ABI